executor/distsql.go

```diff
@@ -745,6 +758,11 @@ func (e *XSelectIndexExec) executeTask(task *lookupTableTask) error {
 		} else {
 			sort.Sort(sorter)
 		}
+		if !e.needColHandle {
```

Comment:
No, rows from `extractRowsFromTableResult` must store handle so there could sort data using it.
But if the executor don't need handle, then we should remove the handle.